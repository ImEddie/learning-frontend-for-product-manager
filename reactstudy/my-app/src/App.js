import React, { Component } from "react";
import Box from "./components/Box.js";
import "./App.scss";

class App extends Component {
  render() {
    return (
      <div className="row">
        <Box></Box>
      </div>
    );
  }
}

export default App;
