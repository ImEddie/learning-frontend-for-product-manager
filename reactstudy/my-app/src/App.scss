.flex {
  display: flex;
  flex-direction: row;
  flex: 1;

  > div {
    margin: 20px;
  }
}

.rows {
  flex: 1;
  border: 1px dashed black;

  > div {
    margin: 10px 0px;
  }
}

.cols {
  flex: 1;
  display: flex;
  flex-direction: row;
  border: 1px dashed black;

  > div {
    flex: 1;
    height: 500px;
    margin: 0px 10px;
  }
}

.center {
  width: 100px;
  height: 100px;
  border: 1px dashed black;
}

.row1 {
  display: flex;

  .box:nth-child(1) {
    flex-grow: 1;
  }

  .box:nth-child(2) {
    flex-grow: 1;
  }

  .box:nth-child(3) {
    flex-grow: 1;
  }
}

.row2 {
  display: flex;

  .box:nth-child(1) {
    flex-grow: 1;
  }

  .box:nth-child(2) {
    flex-grow: 2;
  }

  .box:nth-child(3) {
    flex-grow: 3;
  }
}

.row3 {
  display: flex;

  .box:nth-child(1) {
    width: 100px;
  }

  .box:nth-child(2) {
    flex-grow: 1;
  }

  .box:nth-child(3) {
    flex-grow: 1;
  }
}

.row4 {
  display: flex;
  justify-content: flex-start;
}

.row5 {
  display: flex;
  justify-content: center;
}

.row6 {
  display: flex;
  justify-content: flex-end;
}

.row7 {
  display: flex;
  justify-content: space-between;
}

.row8 {
  display: flex;
  justify-content: space-around;
}

.row9 {
  display: flex;
  justify-content: space-between;
}

.col1 {
  display: flex;
  flex-direction: column;

  .box:nth-child(1) {
    flex-grow: 1;
  }

  .box:nth-child(2) {
    flex-grow: 1;
  }

  .box:nth-child(3) {
    flex-grow: 1;
  }
}

.col2 {
  display: flex;
  flex-direction: column;

  .box:nth-child(1) {
    flex-grow: 1;
  }

  .box:nth-child(2) {
    flex-grow: 2;
  }

  .box:nth-child(3) {
    flex-grow: 3;
  }
}

.col3 {
  display: flex;
  flex-direction: column;

  .box:nth-child(1) {
    height: 100px;
  }

  .box:nth-child(2) {
    flex-grow: 2;
  }

  .box:nth-child(3) {
    flex-grow: 3;
  }
}

.col4 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}

.col5 {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.col6 {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.col7 {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.center {
  display: flex;
  justify-content: center;
  align-items: center;
}
