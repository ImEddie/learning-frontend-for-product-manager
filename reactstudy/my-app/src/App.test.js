import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders button success', () => {
  const tree = renderer
    .create(<App></App>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});