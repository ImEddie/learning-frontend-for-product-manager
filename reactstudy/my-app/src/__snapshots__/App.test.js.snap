exports[`test renders button success 1`] = `
<div>
  <div>
    Button1
  </div>
  <div>
    Button2
  </div>
</div>
`;
